include/swift/Reflection/ReflectionContext.h

```diff
@@ -789,61 +789,150 @@ class ReflectionContext
     }
   }
 
+  bool
```

Comment:
I understand that `projectExistential()` uses a similar API, but if we are adding a new API anyway, why not make this
```
llvm::Optional<std::pair<const TypeRef *, RemoteAddress>> getDynamicTypeAndAddressClassExistential(RemoteAddress object)
```
?

It might also be nice to add a Doxygen comment to function that explains what it does.

Reply:
Sure! I prefer this style as well. 

About the Doxygen comment, I wrote one but it ended up being just repeating the function's name, so I think it might not add much, what do you think?